src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -93,15 +94,26 @@ public override bool Execute()
             return true;
         }
 
-        public string GetCurrentRevisionForBuildName(string buildId)
+        public bool SetVersionAndRevisionFromBuildId(string buildId)
         {
-            Regex regex = new Regex(@"\.(\d+)$");
+            Regex regex = new Regex(@"(\d{8})\.(\d+)$");
```

Comment:
At a future point we should consider making the regex and the dateFormat as a parameter to the task.